injectproxy/routes.go

```diff
@@ -96,6 +98,24 @@ func WithErrorOnReplace() Option {
 	})
 }
 
+func WithExtraHttpHeaders(headers []string) Option {
+	return optionFunc(func(o *options) {
+		o.extraHttpHeaders = make(map[string]string)
+		for _, headerArg := range headers {
+			header, val, found := strings.Cut(headerArg, ":")
+			if found {
+				o.extraHttpHeaders[strings.TrimSpace(header)] = strings.TrimSpace(val)
+			}
+		}
+	})
+}
```

Comment:
I don't see how main would have split the value of the each of the string on the ":" ... I've added that logic into main but I thought it would be cleaner here.